src/plugins/intel_cpu/src/emitters/snippets/aarch64/utils.cpp

```diff
@@ -0,0 +1,176 @@
+    const Xbyak_aarch64::XReg addr_reg = aux_regs[2];  // For address calculations in add_imm
+
+    // Copy pointer to aux register
+    h->mov(aux_reg, ptr_reg);
+
+    // Load the runtime offset from abi_param1 (X0) and add it to the pointer
+    Xbyak_aarch64::XReg abi_param1(0);
+
+    // Load the offset value from the runtime parameter location
+    h->add_imm(temp_reg, abi_param1, runtime_offset, addr_reg);
+    h->ldr(temp_reg, Xbyak_aarch64::ptr(temp_reg));
+
+    h->add(aux_reg, aux_reg, temp_reg);
+
+    // Store the adjusted pointer on stack
+    h->str(aux_reg, Xbyak_aarch64::ptr(h->sp, stack_offset));
+}
+
+void push_ptr_with_static_offset_on_stack(dnnl::impl::cpu::aarch64::jit_generator* h,
+                                          int32_t stack_offset,
+                                          const Xbyak_aarch64::XReg& ptr_reg,
+                                          const std::vector<Xbyak_aarch64::XReg>& aux_regs,
+                                          size_t ptr_offset) {
+    // If there's no static offset, just store the pointer
+    if (ptr_offset == 0) {
+        h->str(ptr_reg, Xbyak_aarch64::ptr(h->sp, stack_offset));
+        return;
+    }
+
+    // Safety assertions as suggested
+    OV_CPU_JIT_EMITTER_ASSERT(aux_regs.size() >= 2, "aux_regs must contain at least 2 registers");
+
+    // Assert that ptr_reg is not in aux_regs
+    for (const auto& reg : aux_regs) {
+        OV_CPU_JIT_EMITTER_ASSERT(reg.getIdx() != ptr_reg.getIdx(), "ptr_reg must not be in aux_regs");
+    }
+
+    // Use safe auxiliary registers from the provided vector
+    const Xbyak_aarch64::XReg temp_reg = aux_regs[0];  // For storing adjusted pointer
+    const Xbyak_aarch64::XReg addr_reg = aux_regs[1];  // For address calculations in add_imm
+
+    // For non-zero offsets, apply the offset and then store
+    h->add_imm(temp_reg, ptr_reg, ptr_offset, addr_reg);
+
+    // Store the adjusted pointer on stack
+    h->str(temp_reg, Xbyak_aarch64::ptr(h->sp, stack_offset));
+}
+
+void push_and_load_ptrs_with_offsets(dnnl::impl::cpu::aarch64::jit_generator* h,
+                                     const std::vector<Xbyak_aarch64::XReg>& mem_ptrs,
+                                     const std::vector<size_t>& memory_offsets,
+                                     const std::vector<size_t>& buffer_ids,
+                                     const std::vector<Xbyak_aarch64::XReg>& aux_regs,
+                                     const std::vector<Xbyak_aarch64::XReg>& load_regs) {
+    // Safety assertions as suggested
+    OV_CPU_JIT_EMITTER_ASSERT(aux_regs.size() >= 3, "aux_regs must contain at least 3 registers");
+
+    // Assert that mem_ptrs do not overlap with aux_regs
+    for (const auto& ptr : mem_ptrs) {
+        for (const auto& aux_reg : aux_regs) {
+            OV_CPU_JIT_EMITTER_ASSERT(ptr.getIdx() != aux_reg.getIdx(), "mem_ptrs must not overlap with aux_regs");
+        }
+    }
```

Comment:
These checks should be only in helpers `push_ptr_with_runtime_offset_on_stack` and `push_ptr_with_static_offset_on_stack` because only these functions need to work with aux registers.

The current method just pass them into these functions. So could you please remove them from here?

Reply:
Yes, removed